### PR TITLE
 [REF] mail: flatten store data sent on network (part 4 - persona)

### PR DESCRIPTION
**Pull request by seb-odoo**
This will reduce data transferred to the client and processed in JS when the same persona was appearing multiple times in various places.

This also makes more clear what is transferred by avoiding nested values.

This is part 4, focusing on persona in general.

Unfortunately `mail_partner_format` itself is not yet converted to store in this PR because `_message_format` needs to be done first, and that will be its own PR (this is the next step).

Formatting code eventually becomes simpler too, by always adding data to the store rather than updating pre-existing dict manually.

Part of task-3605717

https://github.com/odoo/enterprise/pull/65411

### CODE REVIEW

File: addons/mail/static/tests/mock_server/mock_models/res_partner.js [hunk 1]
```diff
@@ -339,37 +339,6 @@ export class ResPartner extends webModels.ResPartner {
         ]).length;
     }
 
-    /**
-     * @param {number} id
-     * @returns {Object[]}
-     */
-    _message_fetch_failed(id) {
```
Comment (seb-odoo) [inline addons/mail/static/tests/mock_server/mock_models/res_partner.js:346]: Note for reviewer: dead code (method is already gone in python)

File: addons/mail/models/mail_thread.py [hunk 2]
```diff
@@ -4526,6 +4527,8 @@ def _get_mail_thread_data(self, request_list):
             res['activities'] = self.with_context(active_test=True).activity_ids.activity_format()
         if 'attachments' in request_list:
             res['attachments'] = self._get_mail_thread_data_attachments()._attachment_format()
+            res["areAttachmentsLoaded"] = True
+            res["isLoadingAttachments"] = False
```
Comment (alexkuhn) [inline addons/mail/models/mail_thread.py:4547]: Don't see where they came from (from the commit this is changed, probably makes sense in global diff)

File: addons/mail/static/tests/mock_server/mail_mock_server.js [hunk 3]
```diff
@@ -358,7 +349,7 @@ async function discuss_channel_mute(request) {
     }
     DiscussChannelMember.write([member.id], { mute_until_dt });
     const channel_data = {
-        id: member.channel_id[0],
```
Comment (alexkuhn) [inline addons/mail/static/tests/mock_server/mail_mock_server.js:361]: oops :)

Comment (seb-odoo) [reply addons/mail/static/tests/mock_server/mail_mock_server.js:361]: `(search_)read` was replaced `_filter` in `_find_or_create_member_for_self`

(`read` does the `id, display_name` syntax)

**Review by alexkuhn — APPROVED** [review]
@robodoo r+ delegate+ rebase-ff override=ci/security
safe insert of Message and Activity data, whose html field (`body` and `note` respectively) are sanitised

**Pull request closed without merging**